crates/state-asm/src/analyse_reads.rs

```diff
@@ -0,0 +1,63 @@
+use crate::StateRead as Op;
+use bitflags::bitflags;
+
+/// Flags to indicate the type of state read operations.
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+pub struct StateReadFlags(u32);
+
+bitflags! {
+    impl StateReadFlags: u32 {
+        /// No state read operations.
+        const None = 0b00000000;
```

Comment:
I think you just mark a `1` on the bit you want to use for each const so maybe None doesn't make sense?